extension/app/ts/background/iconHandler.ts

```diff
@@ -24,26 +26,68 @@ export function updateExtensionIcon(port: browser.runtime.Port) {
-			return setIcon(port.sender.tab.id, ICON_ACCESS_DENIED, `The access for ${ origin } has been DENIED!`)
+			return setInterceptorIcon(port.sender.tab.id, ICON_ACCESS_DENIED, `The access for ${ origin } has been DENIED!`)
 		}
-		return setIcon(port.sender.tab.id, ICON_ACCESS_DENIED, `The access to ${ getAddressMetaData(activeAddress, window.interceptor.settings?.userAddressBook).name } for ${ origin } has been DENIED!`)
+		return setInterceptorIcon(port.sender.tab.id, ICON_ACCESS_DENIED, `The access to ${ getAddressMetaData(activeAddress, window.interceptor.settings?.userAddressBook).name } for ${ origin } has been DENIED!`)
 	}
-	if (window.interceptor.settings?.simulationMode) return setIcon(port.sender.tab.id, ICON_SIMULATING, `The Interceptor simulates your sent transactions.`)
-	if (!isSupportedChain(window.interceptor.settings.activeChain.toString())) return setIcon(port.sender.tab.id, ICON_SIGNING_NOT_SUPPORTED, `Interceptor is on an unsupported network and simulation mode is disabled.`)
+	if (window.interceptor.settings?.simulationMode) return setInterceptorIcon(port.sender.tab.id, ICON_SIMULATING, `The Interceptor simulates your sent transactions.`)
+	if (!isSupportedChain(window.interceptor.settings.activeChain.toString())) return setInterceptorIcon(port.sender.tab.id, ICON_SIGNING_NOT_SUPPORTED, `Interceptor is on an unsupported network and simulation mode is disabled.`)
 
-	return setIcon(port.sender.tab.id, ICON_SIGNING, `The Interceptor forwards your transactions to ${ getSignerName(window.interceptor.signerName) } once sent.`)
+	return setInterceptorIcon(port.sender.tab.id, ICON_SIGNING, `The Interceptor forwards your transactions to ${ getSignerName(window.interceptor.signerName) } once sent.`)
 }
 
 export async function updateExtensionBadge() {
 	if (!window.interceptor.settings) return
 	const count = window.interceptor.settings.pendingAccessRequests.length
 	return await browser.browserAction.setBadgeText( { text: count === 0 ? '' : count.toString() } )
 }
+
+export async function retrieveWebsiteTabIcon(tabId: number | undefined) {
+	if ( tabId === undefined) return undefined
+
+	// wait for the tab to be fully loaded
+	const waitForLoadedFuture = new Future<void>
+	const waitForLoaded = async () => {
+		try {
+			const listener = function listener(tabIdUpdated: number, info: browser.tabs._OnUpdatedChangeInfo) {
+				if (info.status === 'complete' && tabId === tabIdUpdated) {
+					browser.tabs.onUpdated.removeListener(listener)
+					return waitForLoadedFuture.resolve()
+				}
+			}
+			browser.tabs.onUpdated.addListener(listener)
+			if( (await browser.tabs.get(tabId)).status === 'complete') {
+				browser.tabs.onUpdated.removeListener(listener)
+				return waitForLoadedFuture.resolve()
+			}
+			return waitForLoadedFuture.resolve()
+		} catch(error) {
+			if (error instanceof Error) return waitForLoadedFuture.reject(error)
+			return waitForLoadedFuture.reject(new Error('Unknown error'))
+		}
+	}
+
+	waitForLoaded()
```

Comment:
No need for the intermediate function here.  
```suggestion
	try {
		const listener = function listener(tabIdUpdated: number, info: browser.tabs._OnUpdatedChangeInfo) {
			if (info.status === 'complete' && tabId === tabIdUpdated) {
				browser.tabs.onUpdated.removeListener(listener)
				return waitForLoadedFuture.resolve()
			}
		}
		browser.tabs.onUpdated.addListener(listener)
		if( (await browser.tabs.get(tabId)).status === 'complete') {
			browser.tabs.onUpdated.removeListener(listener)
			return waitForLoadedFuture.resolve()
		}
		return waitForLoadedFuture.resolve()
	} catch(error) {
		if (error instanceof Error) return waitForLoadedFuture.reject(error)
		return waitForLoadedFuture.reject(new Error('Unknown error'))
	}
```